File: javascript/anagram/anagram.js
var Anagram = function(seed) {

  var cleaner = function(start) {
    return start.toLowerCase().split('').sort().join('');
  };

  var target = cleaner(seed);


  this.matches = function(candidates) {
    if (typeof candidates == 'string') {
      candidates = Array.prototype.slice.call(arguments);
    }
    return candidates.filter(function(candidate) {
      return target == cleaner(candidate) &&
        seed.toLowerCase() != candidate.toLowerCase();
    });
  };
};

module.exports = Anagram;

File: ruby/anagram/anagram.rb
class Anagram
  attr_reader :seed

  def initialize(s)
    @seed = s.downcase
  end

  def match(candidates)
    target = munge(seed)
    candidates.select{|c| munge(c) == target && c.downcase != seed }
  end

  private

  def munge(s)
    s.downcase.chars.sort.join
  end
end
